lib/WebpackConfig.js

```diff
@@ -749,9 +750,21 @@ class WebpackConfig {
             if (!(key in this.vueOptions)) {
                 throw new Error(`"${key}" is not a valid key for enableVueLoader(). Valid keys: ${Object.keys(this.vueOptions).join(', ')}.`);
             }
+
+            if (key === 'version') {
+                const validVersions = [2, 3];
+                if (!validVersions.includes(vueOptions.version)) {
+                    throw new Error(`"${vueOptions.version}" is not a valid value for the "version" option passed to enableVueLoader(). Valid versions are: ${validVersions.join(', ')}.`);
+                }
+            }
+
+            this.vueOptions[key] = vueOptions[key];
         }
 
-        this.vueOptions = vueOptions;
+        // useJsx and vue 3 are not currently supported by Encore
+        if (this.vueOptions.useJsx && this.vueOptions.version === 3) {
+            throw new Error('Setting both "useJsx: true" and "version: 3" for enableVueLoader() is not currently supported. Please use version: 2 or disable useJsx.');
+        }
```

Comment:
It appears that the babel preset that enables this doesn't work yet - https://github.com/vuejs/jsx/issues/106